src/command/utils.rs

```diff
@@ -8,8 +8,34 @@ use walkdir::WalkDir;
 
 /// If an explicit path is given, then use it, otherwise assume the current
 /// directory is the crate path.
-pub fn set_crate_path(path: Option<PathBuf>) -> Result<PathBuf, failure::Error> {
-    Ok(path.unwrap_or_else(|| PathBuf::from(".")))
+pub fn get_crate_path(path: Option<PathBuf>) -> Result<PathBuf, failure::Error> {
+    match path {
+        Some(p) => Ok(p),
+        None => find_manifest_from_cwd(),
+    }
+}
+
+/// Search up the path for the manifest file from the current working directory
+/// If we don't find the manifest file then return back the current working directory
+/// to provide the appropriate error
+fn find_manifest_from_cwd() -> Result<PathBuf, failure::Error> {
+    let cwd = PathBuf::from(".");
+    let parent_path = cwd.canonicalize()?;
```

Comment:
I'm a little hesitant to add `canonicalize` calls here, since we have historically had issues with canonicalized paths on windows (and in non-english locales?) that I never fully understood the root causes for.

If this isn't required for correctness (I don't think any APIs we use require canonical paths, and work just fine with relative paths), then I'd like to remove the `canonicalize` call. Or maybe @ashleygwilliams or @alexcrichton understood the bug we hit better than I do.

Sorry I missed this in the first round of review!

Reply:
I think this shoudl be fixable by basically just using `env::current_dir()` rather than `canonicalize` which should return an appropriate absolute path

Reply:
Before refactoring I did need to use canonicalize - the problem was that a relative path coming into the function (e.g. PathBuff::from(".") that was the default) - isn't absolute so parent_path.parent() returns None even if you are not in the root directory. I needed to canonicalize it to ensure I had an absolute path.

However the logic changed after migrating the code into the the get_crate_path as now we take the current working directory and then work back from there. So thanks @alexcrichton using env::current_dir() gives the absolute path  needed. Looking back at the output on Windows of the cwd after it was canonicalized it had a whole bunch of escaping included - compared to the absolute path.